apps/home/views.py

```diff
@@ -3,43 +3,51 @@
 from ..accounts.models import UserProfile
 from ..fhir.bluebutton.models import Crosswalk
 from django.views.decorators.cache import never_cache
-
-
-def home(request):
-    template = "index.html"
-    if request.user.is_authenticated:
-        return authenticated_home(request)
-    return render(request, template, {})
-
-
-@never_cache
-def authenticated_home(request):
-
-    template = 'authenticated-home.html'
-    if request.user.is_authenticated:
-        name = _('Authenticated Home')
-        try:
-            profile = UserProfile.objects.get(user=request.user)
-        except UserProfile.DoesNotExist:
-            profile = None
-        try:
-            crosswalk = Crosswalk.objects.get(user=request.user)
-        except Crosswalk.DoesNotExist:
-            crosswalk = None
-
-        if crosswalk is None:
-            fhir_id = '0'
+from oauth2_provider.models import get_application_model
+from django.views.generic.base import TemplateView
+from django.shortcuts import redirect
+from django.contrib.auth.mixins import LoginRequiredMixin
+
+Application = get_application_model()
+
+class HomeView(TemplateView):
+    template_name = "index.html"
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return redirect('/home')
+        return super().get(request, *args, **kwargs)
+
+
+class AuthenticatedHomeView(LoginRequiredMixin, TemplateView):
+    template_name = 'authenticated-home.html'
+
+    def get_context_data(self, **kwargs):
+        request = self.request
+        if request.user.is_authenticated:
+            name = _('Authenticated Home')
+            try:
+                profile = UserProfile.objects.get(user=request.user)
+            except UserProfile.DoesNotExist:
+                profile = None
+            try:
+                crosswalk = Crosswalk.objects.get(user=request.user)
+            except Crosswalk.DoesNotExist:
+                crosswalk = None
+
+            if crosswalk is None:
+                fhir_id = '0'
+            else:
+                fhir_id = crosswalk.fhir_id
+
+            if fhir_id == '':
+                fhir_id = '0'
+            # this is a GET
+            context = {'name': name, 'profile': profile,
+                    'crosswalk': crosswalk,
+                    'fhir_id': fhir_id,
+                    'applications': Application.objects.filter(user=request.user)}
         else:
```

Comment:
i believe this is unreachable code.

Reply:
Is this just something you noticed that you want to change?

Reply:
consider it part of the hunt for "dead code". I'm not sure if it was dead before the updates but I suspect it was.

Reply:
So, which section should be removed? Just this?
```py
else:
      name = ('home')
      context = {'name': name}
```

Reply:
that sounds right

Reply:
Okay. Just removed this `else` statement and committed. I clicked around and it seemed like all functionality was still there.